oi/type_graph/NameGen.cpp

```diff
@@ -15,6 +15,12 @@
  */
 #include "NameGen.h"
 
+#include <algorithm>
+#include <cstddef>
+#include <memory>
+#include <optional>
```

Comment:
I also can't see how a few of these headers are needed for this class

Reply:
Optional is used when you dereference `param.value`:
https://github.com/facebookexperimental/object-introspection/blob/55d1b1300960e4d096f7df7fdfe32977cfa1ecbc/oi/type_graph/NameGen.cpp#L120

The `<memory>` one I'm a bit stumped by. It wants it in loads of places because "// for allocator_traits<>::value_type". I think it includes everything you need for a templated function's implementation, so perhaps string is using it.